scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -139,16 +154,35 @@ namespace <%= namespace %> {
                 CurrentCheckout = merger.Merge(CurrentCheckout, checkout);    
             }
 
+            if (userErrors.Count > 0) {
+                _UserErrors = userErrors;
+            } else {
+                _UserErrors = null;
+            }
+
             UpdateLineItemFromCheckout(CurrentCheckout);
+
+            return _UserErrors == null;
         }
 
         private void UpdateLineItemFromCheckout(Checkout checkout) {
-            // sometimes we may not query line items for instance when polling is being performed
-            try {
-                List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
+            if (checkout != null) {
+                // sometimes we may not query line items for instance when polling is being performed
+                try {
+                    List<CheckoutLineItem> lineItems = (List<CheckoutLineItem>) checkout.lineItems();
+
+                    LineItems.UpdateLineItemsFromCheckoutLineItems(lineItems);
+                } catch(NoQueryException exception) {}
+            }
+        }
+
+        private void HandleUserError(Action<ShopifyError> callback) {
+            ShopifyError error = new ShopifyError(
+                ShopifyError.ErrorType.UserError,
+                "There were issues with some of the fields sent. Checkout `cart.UserErrors`"
```

Comment:
Nit: Super minor but maybe reword 'Checkout' to 'See' since Checkout is a bit overloaded 🙃